src/Core/Utilities/CoreHelpers.cs

```diff
@@ -264,5 +264,38 @@ public static T Clone<T>(T obj)
         {
             return JsonConvert.DeserializeObject<T>(JsonConvert.SerializeObject(obj));
         }
+
+        public static string TextColorFromBgColor(string hexColor, int threshold = 166)
+        {
+            var bgColorHexRgb = hexColor.Substring(0, 1) == "#" ? hexColor.Substring(3, 6) : hexColor.Substring(2, 6);
+            var r = Convert.ToUInt32(bgColorHexRgb.Substring(0, 2), 16);
+            var g = Convert.ToUInt32(bgColorHexRgb.Substring(2, 2), 16);
+            var b = Convert.ToUInt32(bgColorHexRgb.Substring(4, 2), 16);
+            var blackColor = "#ff000000";
+            var whiteColor = "#ffffffff";
+            var comp = r * 0.299 + g * 0.587 + b * 0.114;
+            return r * 0.299 + g * 0.587 + b * 0.114 > threshold ? blackColor : whiteColor;
+        }
```

Comment:
This can be simply written by:
```c#
var bgColor = (Color)new ColorConverter().ConvertFromString(hexColor);
var luminance = bgColor.R * 0.299 + bgColor.G * 0.587 + bgColor.B * 0.114;
return luminance > threshold ? "#ff000000" : "#ffffffff";
```

Reply:
This required a null check before casting - let me know if you're ok with the adjustment